crates/typos-cli/src/bin/typos-cli/report.rs

```diff
@@ -1,12 +1,15 @@
 #![allow(clippy::needless_update)]
 
-use std::io::Write as _;
-use std::sync::atomic;
+use std::cell::RefCell;
+use std::io::{Error, Write as _};
+use std::sync::{atomic, Mutex};
 
 use anstream::stdout;
+use serde_sarif::sarif;
 use unicode_width::UnicodeWidthStr;
+use serde_sarif::sarif::{MessageBuilder, ResultLevel};
```

Comment:
nit: Generally, I prefer to minimize what types get brought in

Reply:
i just have no idea

Reply:
Ok